### ISSUE TITLE
Missing transpose on loss function?

### ISSUE THREAD

**Issue by renato145** [UK-Digital-Heart-Project/4Dsurvival#1]
Hi,
I have noticed that in the negative likelihood loss functions there is no transpose on line 39 (https://github.com/UK-Digital-Heart-Project/4Dsurvival/blob/master/code/trainDL.py#L39). This is the code deepsurv implementation: https://github.com/jaredleekatzman/DeepSurv/blob/master/deepsurv/deep_surv.py#L165.

**Comment by ghalib-bello** [comment]
Hi,
Since the parameters for our `_negative_log_likelihood` function (`E` and `risk`) are both (n x 1) tensors, the suggested transpose operation is not needed. Because both `E` and `risk` are (n x 1), the variable defined on line 39 (`uncensored_likelihood`) would also be (n x 1). If we instead use a transpose operation as suggested, `uncensored_likelihood` would then become (n x n), which is incorrect. The deepsurv code you cite is in Theano (not Tensorflow), and the `E` tensor in that implementation is of dimension (n), not (n x 1).
For more details on our `_negative_log_likelihood` function, see Section 4 of [code/trainDL.ipynb](https://github.com/UK-Digital-Heart-Project/4Dsurvival/blob/master/code/trainDL.ipynb)

**Comment by renato145** [comment]
That clear some things out for implementing the loss on Pytorch, thank you!
